src/spdx/parser/tagvalue/lexer.py

```diff
@@ -0,0 +1,222 @@
+        "PackageLicenseInfoFromFiles": "PKG_LICENSE_INFO",
+        "PackageLicenseComments": "PKG_LICENSE_COMMENT",
+        "PackageCopyrightText": "PKG_COPYRIGHT_TEXT",
+        "PackageHomePage": "PKG_HOMEPAGE",
+        "ExternalRef": "PKG_EXTERNAL_REF",
+        "ExternalRefComment": "PKG_EXTERNAL_REF_COMMENT",
+        "PackageAttributionText": "PKG_ATTRIBUTION_TEXT",
+        "PrimaryPackagePurpose": "PRIMARY_PACKAGE_PURPOSE",
+        "BuiltDate": "BUILT_DATE",
+        "ReleaseDate": "RELEASE_DATE",
+        "ValidUntilDate": "VALID_UNTIL_DATE",
+        # File fields
+        "FileName": "FILE_NAME",
+        "FileType": "FILE_TYPE",
+        "FileChecksum": "FILE_CHECKSUM",
+        "LicenseConcluded": "FILE_LICENSE_CONCLUDED",
+        "LicenseInfoInFile": "FILE_LICENSE_INFO",
+        "FileCopyrightText": "FILE_COPYRIGHT_TEXT",
+        "LicenseComments": "FILE_LICENSE_COMMENT",
+        "FileComment": "FILE_COMMENT",
+        "FileNotice": "FILE_NOTICE",
+        "FileContributor": "FILE_CONTRIBUTOR",
+        "FileAttributionText": "FILE_ATTRIBUTION_TEXT",
+        # ExtractedLicensingInfo fields
+        "LicenseID": "LICENSE_ID",
+        "ExtractedText": "LICENSE_TEXT",
+        "LicenseName": "LICENSE_NAME",
+        "LicenseCrossReference": "LICENSE_CROSS_REF",
+        "LicenseComment": "LICENSE_COMMENT",
+        # Snippet fields
+        "SnippetSPDXID": "SNIPPET_SPDX_ID",
+        "SnippetName": "SNIPPET_NAME",
+        "SnippetComment": "SNIPPET_COMMENT",
+        "SnippetCopyrightText": "SNIPPET_COPYRIGHT_TEXT",
+        "SnippetLicenseComments": "SNIPPET_LICENSE_COMMENT",
+        "SnippetFromFileSPDXID": "SNIPPET_FILE_SPDXID",
+        "SnippetLicenseConcluded": "SNIPPET_LICENSE_CONCLUDED",
+        "LicenseInfoInSnippet": "SNIPPET_LICENSE_INFO",
+        "SnippetAttributionText": "SNIPPET_ATTRIBUTION_TEXT",
+        "SnippetByteRange": "SNIPPET_BYTE_RANGE",
+        "SnippetLineRange": "SNIPPET_LINE_RANGE",
+        # Common fields
+        "NOASSERTION": "NO_ASSERTION",
+        "NONE": "NONE"
+    }
+    states = (("text", "exclusive"),)
+
+    tokens = [
+                 "TEXT",
+                 "TOOL_VALUE",
+                 "UNKNOWN_TAG",
+                 "ORGANIZATION_VALUE",
+                 "PERSON_VALUE",
+                 "DATE",
+                 "LINE",
+                 "CHECKSUM",
+                 "EXT_DOC_REF_ID",
+                 "EXT_DOC_URI",
+                 "EXT_DOC_REF_CHECKSUM",
+             ] + list(reserved.values())
+
+    def __init__(self):
+        self.lexer = None
+
+    @TOKEN(r":\s*<text>")
+    def t_text(self, t):
+        t.lexer.text_start = t.lexer.lexpos - len("<text>")
+        t.lexer.begin("text")
+
+    @TOKEN(r"</text>\s*")
+    def t_text_end(self, t):
+        t.type = "TEXT"
+        t.value = t.lexer.lexdata[t.lexer.text_start: t.lexer.lexpos]
+        t.lexer.lineno += t.value.count("\n") if "\n" in t.value else 1
```

Comment:
```suggestion
        t.lexer.lineno += t.value.count("\n") + 1
```
This way, if the text contains a new line, the line number increases by 2. That's my understanding of how this should work

Reply:
The linebreak after the closing `</text>` file also increase the line number by one, so this solution would lead to a mismatch of lines.

Reply:
The original code was correct and the test which I thought implied this adaption here was simply missing a comma. :woman_facepalming:

Reply:
Nice :D